app/code/community/Bolt/Boltpay/Model/FeatureSwitch.php

```diff
@@ -50,12 +94,109 @@ public function updateFeatureSwitches()
         $switches = array();
         foreach ($switchesData as $switch) {
             $switches[$switch->name] = (object)array(
-                'value' => $switch->value,
-                'defaultValue' => $switch->defaultValue,
-                'rolloutPercentage' => $switch->rolloutPercentage
+                self::VAL_KEY => $switch->value,
+                self::DEFAULT_VAL_KEY => $switch->defaultValue,
+                self::ROLLOUT_KEY => $switch->rolloutPercentage
             );
         }
-        Mage::getModel('core/config')->saveConfig('payment/boltpay/featureSwitches', json_encode($switches));
+        $this->switches = $switches;
+        Mage::getModel('core/config')->saveConfig('payment/boltpay/featureSwitches', serialize($switches));
         Mage::getModel('core/config')->cleanCache();
     }
+
+    /**
+     * Read feature switches from database if we didn't it before
+     */
+    protected function readSwitches()
+    {
+        if (isset($this->switches)) {
+            return;
+        }
+        $this->switches = unserialize(Mage::getStoreConfig('payment/boltpay/featureSwitches'));
+    }
+
+    /**
+     * Get unique ID from cookie or generate it and save it in cookie and session
+     * @return string
+     */
+    protected function getUniqueUserId()
+    {
+        $boltFeatureSwitchId = Mage::getSingleton('core/cookie')->get(SELF::COOKIE_NAME);
+        if (!$boltFeatureSwitchId) {
+            $boltFeatureSwitchId = uniqid("BFS", true);
+            $boltFeatureSwitchId = Mage::getSingleton('core/cookie')->set(SELF::COOKIE_NAME, $boltFeatureSwitchId, true);
+        }
+        return $boltFeatureSwitchId;
+    }
+
+    /**
+     * This method returns if a feature switch is enabled for a user.
+     * The way this is computed is as follows:
```

Comment:
Missing some comment? What is the way this is computed?

Reply:
Added